android-components/components/feature/search/src/main/java/mozilla/components/feature/search/telemetry/SerpTelemetryFetcher.kt

```diff
@@ -0,0 +1,122 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.search.telemetry
+
+import mozilla.appservices.remotesettings.RemoteSettingsResponse
+import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.ktx.android.org.json.asSequence
+import mozilla.components.support.ktx.android.org.json.toList
+import mozilla.components.support.remotesettings.RemoteSettingsClient
+import mozilla.components.support.remotesettings.RemoteSettingsResult
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.File
+
+internal const val REMOTE_ENDPOINT_URL = "https://firefox.settings.services.mozilla.com"
+internal const val REMOTE_ENDPOINT_BUCKET_NAME = "main"
+
+/**
+ * Parse SERP Telemetry json from remote config.
+ */
+class SerpTelemetryRepository(
+    rootStorageDirectory: File,
+    collectionName: String,
+    serverUrl: String = REMOTE_ENDPOINT_URL,
+    bucketName: String = REMOTE_ENDPOINT_BUCKET_NAME,
+) {
+    val logger = Logger("SerpTelemetryRepository")
+    private val remoteSettingsClient = RemoteSettingsClient(
+        serverUrl = serverUrl,
+        bucketName = bucketName,
+        collectionName = collectionName,
+        storageRootDirectory = rootStorageDirectory,
+    )
+
+    /**
+     * Writes data to local cache.
+     */
+    suspend fun writeToCache(records: RemoteSettingsResponse) {
+        remoteSettingsClient.write(records)
+    }
+
+    /**
+     * Parses local json response.
+     */
+    fun parseLocalData(jsonObject: JSONObject): List<SearchProviderModel> {
+        val localesConfig = jsonObject.getJSONArray("data")
+        return localesConfig.asSequence()
+            .mapNotNull {
+                (it as JSONObject).toSearchModelProvider()
+            }
+            .toList()
+    }
+
+    /**
+     * Returns data from remote servers.
+     */
+    suspend fun fetchRemoteResponse(): RemoteSettingsResponse {
+        val result = remoteSettingsClient.fetch()
+        return if (result is RemoteSettingsResult.Success) {
+            result.response
+        } else {
+            RemoteSettingsResponse(emptyList(), 0u)
+        }
+    }
+
+    /**
+     * Returns search providers from local cache.
+     */
+    suspend fun fetchProvidersFromCache(): Pair<ULong, List<SearchProviderModel>> {
```

Comment:
nit: I think we should stick with `fetch` for remote and `load` for local
```suggestion
    suspend fun loadProvidersFromCache(): Pair<ULong, List<SearchProviderModel>> {
```